### DISCUSSION TITLE
Markdown support

### DISCUSSION THREAD

**Discussion by jayvdb** [jeertmans/languagetool-rust#116] [Ideas]
afaics, this is the best (only?) Rust cli tool for LanguageTool , however it is missing one feature which JS tool https://github.com/caderek/gramma and https://github.com/twilio-labs/languagetool-cli has : markdown support.

I've played with gramma , and it works quite well, eliminating a lot of false positives for markdown files.

**Comment by jeertmans** [comment]
Hello @jayvdb!

Indeed, this would be great to provide built-in support for Markdown.

The best would probably to re-use a markdown-parser library (maybe a pull parser for performances?) and just turn the "token" into a vec of [`DataAnnotation`](https://docs.rs/languagetool-rust/latest/languagetool_rust/check/struct.DataAnnotation.html).

Unfortunately, I do not have much time myself to develop additional features for LTRS, and I only really focused on the API bindings.

If you (or anyone else) wants to give it a try, please do and I'll happily review any PR :-)
